src/class-autoloader.php

```diff
@@ -11,6 +11,96 @@
  * WordPress Autoload Generator
  */
 class Autoloader {
+	/**
+	 * Namespace to autoload.
+	 *
+	 * @var string
+	 */
+	public string $namespace;
+
+	/**
+	 * Root path of the namespace to load from.
+	 *
+	 * @var string
+	 */
+	public string $root_path;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string $namespace Namespace to register.
+	 * @param string $root_path Root path of the namespace.
+	 */
+	public function __construct( string $namespace, string $root_path ) {
+		$this->namespace = $namespace;
+		$this->root_path = $root_path;
+	}
+
+	/**
+	 * Register the autoloader.
+	 */
+	public function register() {
+		spl_autoload_register( $this );
+	}
+
+	/**
+	 * Unregister the autoloader.
+	 */
+	public function unregister() {
+		spl_autoload_unregister( $this );
+	}
+
+	/**
+	 * Invoke method of the class.
+	 *
+	 * @param string $classname Class being autoloaded.
+	 */
+	public function __invoke( string $classname ) {
+		// Ignore if the base namespace doesn't match.
+		if ( 0 !== \strpos( $classname, $this->namespace ) ) {
```

Comment:
would this match if one classname was a subset of the other:
`\MyNameSpace\MyClassName` and `\MyNameSpace\MyClassNameFeature` ?

Reply:
Yes it should!